metaflow/plugins/aip/argo_utils.py

```diff
@@ -24,8 +24,41 @@ def __init__(self, kubernetes_namespace: str = KUBERNETES_NAMESPACE):
                 Required as the defaults provided in the ArgoClient is usually not what customers desire.
                 TODO: This namespace can be default to the current namespace if the script is ran within a cluster.
         """
+        logger.info(f"initiating ArgoHelper for Namespace: {kubernetes_namespace}")
         self._client = ArgoClient(namespace=kubernetes_namespace)
 
+    def stop_workflow(self, workflow_name: str) -> None:
+        """
+        Stop a workflow but still run exit handlers.
```

Comment:
* It'd be useful to the end user to comment when `stop_workflow` runs the exit handler?  Does it run it on every `stop_workflow` or only if the currently running step fails then the exit handler is run?
* And also that the exit handler is necessary to send email to OpsGenie for the OpsGenie alert.

Reply:
I'm not sure I understand the question. I'm happy to chat over zoom to make sure I understand.

To try to answer here, `stop_workflow()` stops any currently-running steps from running, and prevents all future steps from running, except the exit handlers, which run immediately.

Are you saying that I should add logs for the customer?

Reply:
Thanks for clearing this up over zoom! I will update the function descriptions to clarify what happens to the workflow (workflow status, whether current step finishes, exit handlers, etc.)

Reply:
I updated the function descriptions for clarity.

```
    def stop_workflow(self, workflow_name: str) -> None:
        """
        Permanently stops workflow with status 'Failed'.
        Currently running steps do NOT finish running.
        Exit handlers are allowed to run.
        Args:
            workflow_name: Name of the workflow to stop.
        """

    def terminate_workflow(self, workflow_name: str) -> None:
        """
        Permanently stops workflow with status 'Failed'.
        Currently running steps do NOT finish running.
        Exit handlers are NOT allowed to run.
        Args:
            workflow_name: Name of the workflow to terminate.
        """
```

Reply:
Looks good!

Adding something akin to the following would be more explicit on user impact in their language:
```
Exit handlers are allowed to run including the `notify-email-exit-handler` which will send an email to OpsGenie to raise an OpsGenie alert.
```

Similarly for the `terminate_workflow` comment, explicitly stating that the `notify-email-exit-handler`  will not run meaning that no OpsGenie alert is raised.

Reply:
Thanks! I updated the descriptions to reflect your comment.